src/hooks/useLoginForm.ts

```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react'
+
+export function useLoginForm() {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [emailError, setEmailError] = useState('')
+  const [passwordError, setPasswordError] = useState('')
+  const [emailTouched, setEmailTouched] = useState(false)
+  const [passwordTouched, setPasswordTouched] = useState(false)
+
+  const isEmailValid = (value: string) =>
+    /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value)
+
+  const isPasswordValid = (value: string) => value.length >= 8
+
+  const validateEmail = (value: string) => {
+    setEmailTouched(true)
+    if (!value) {
+      setEmailError('ID를 입력해주세요.')
+    } else if (!isEmailValid(value)) {
+      setEmailError('ID는 이메일 형식으로 입력해주세요.')
+    } else {
+      setEmailError('')
+    }
+  }
+
+  const validatePassword = (value: string) => {
+    setPasswordTouched(true)
+    if (!value) {
+      setPasswordError('PW를 입력해주세요.')
+    } else if (!isPasswordValid(value)) {
+      setPasswordError('PW는 최소 8글자 이상이어야 합니다.')
+    } else {
+      setPasswordError('')
+    }
+  }
+
+  const isFormValid = isEmailValid(email) && isPasswordValid(password)
+
+  return {
+    email,
+    setEmail,
+    password,
+    setPassword,
+    emailError,
+    passwordError,
+    emailTouched,
+    passwordTouched,
+    validateEmail,
+    validatePassword,
+    isFormValid,
+    setEmailTouched,
+    setPasswordTouched,
+  }
```

Comment:
이렇게 하나의 객체로 넘기면 가독성이 굉장히 안 좋아진답니다 ㅎㅎ
namespace로 묶어주면 어떨까요?

```suggestion
  return {
    email: {
      value: email,
      change: setEmail,
      error: emailError,
      onBlur: () => setEmailTouched(true),
      validate: vadateEMail
    },
    password: {
      value: password,
      change: setPassword,
      error: passwordError,
      onBlur: () => setPasswordTouched(true),
      validate: validatePassword,
    },
    validForm: isFormValid,
  }
```

이렇게요!